cavern/src/main/java/org/opencadc/cavern/pkg/CavernPackageRunner.java

```diff
@@ -0,0 +1,297 @@
+            log.debug("targetPath isDirectory: " + Files.isDirectory(targetPath));
+            log.debug("targetPath isRegularFile: " + Files.isRegularFile(targetPath));
+
+            // This is needed to set the relative path of everything in ManifestList
+            // packageParentPath will be pruned off the file paths returned
+            // from walkFileTree
+            Path packageParentPath = targetPath.getParent();
+            if (packageParentPath == null) {
+                // At the verly least the CANFAR root path should be
+                // passed in so it can be pruned off the file paths
+                packageParentPath = rootPath;
+            }
+
+            if (Files.isRegularFile(targetPath)) {
+                // links and files
+                log.debug("file item found (no tree walk) " + targetPath);
+                addToPkgList(targetPath, packageParentPath, packageItems);
+            } else {
+                // Build the full path to files, including canfar_root
+                log.debug("not regular file - will walk it " + targetPath);
+
+                ManifestVisitor mv = new ManifestVisitor(rootPath);
+                Files.walkFileTree(targetPath, EnumSet.of(FileVisitOption.FOLLOW_LINKS), Integer.MAX_VALUE, mv);
+                log.debug("manifest list size: " + mv.getManifestList().size());
+
+                for (Path p : mv.getManifestList()) {
+                    log.debug("path: " + p);
+                    // p doesn't have VOS_FILESYSTEM_ROOT or rootPath attached.
+                    // system won't find it.
+                    Path fullPath = rootPath.resolve(p);
+
+                    log.debug("file item found after tree walk: " + fullPath);
+                    addToPkgList(fullPath, packageParentPath, packageItems);
+                }
+            }
+
+        }
+        log.debug("target list: " + targetList.toString());
+        log.debug("getItems returning");
+        return packageItems.iterator();
+    }
+
+
+    /**
+     * Add the targetPath to the package list (piList).
+     * @param targetPath - Path of target to add to PackageItem list
+     * @param packageParentPath - Path to root of the package being built
+     * @param piList - list of PackageItem being added to
+     */
+    private void addToPkgList(Path targetPath, Path packageParentPath, List<PackageItem> piList) {
+        log.debug("adding path to list: " + targetPath.toString());
+        try {
+            // Generate a file:// URL using the targetPath passed in
+            // Assumption here is canfar_root is well known, so there isn't a need
+            // to compare the path parts - it has been added to anything in the ManifestList,
+            // so subpath using count of parts is safe here
+            URL fileURL = new URL("file://" + targetPath.toString());
+            log.debug("file uri: " + fileURL.toString());
+
+            Path relativePath;
```

Comment:
Move the declaration down to where it is assigned.

Reply:
ok